File: perf/bench/bitops/bitops.swift
func popcount(_ x0: UInt64) -> Int64 {
    var x = x0
    var n: Int64 = 0
    while x != 0 { x = x & (x - 1); n += 1 }
    return n
}
func mix(_ h0: UInt64) -> UInt64 {
    var h = h0
    h = h ^ (h >> 33)
    h = h &* 18397679294719823053
    h = h ^ (h >> 29)
    return h
}
var total: Int64 = 0
var h: UInt64 = 12345
for _ in 0..<20000000 { h = mix(h); total += popcount(h) }
print(total)

File: perf/bench/bitops/bitops.go
package main

import "fmt"

func popcount(x uint64) int64 {
	var n int64 = 0
	for x != 0 {
		x = x & (x - 1)
		n++
	}
	return n
}

func mix(h uint64) uint64 {
	h = h ^ (h >> 33)
	h = h * 18397679294719823053
	h = h ^ (h >> 29)
	return h
}

func main() {
	var total int64 = 0
	var h uint64 = 12345
	for i := 0; i < 20000000; i++ {
		h = mix(h)
		total += popcount(h)
	}
	fmt.Println(total)
}
